CHANGELOG.md

```diff
@@ -6,6 +6,76 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 (modification: no type change headlines) and this project adheres to
 [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [4.0.0] - 2020-04-17
+
+This release introduces a major API upgrade from callbacks to Promises.
+
+Example using async/await syntax:
+
+```typescript
+import { BaseTrie as Trie } from 'merkle-patricia-tree'
+
+const trie = new Trie()
+await trie.put(Buffer.from('test'), Buffer.from('one'))
+const value = await trie.get(Buffer.from('test'))
```

Comment:
What do you think of including these inside an async function, so that the code doesn't return an error when pasted as-is by users? A console.log might also be helpful for new users.
```suggestion
async function test() {
    await trie.put(Buffer.from('test'), Buffer.from('one'))
    const value = await trie.get(Buffer.from('test'))
    console.log(value) // 'one'
}
```

Reply:
sure, thanks :)